.github/actions/github-check-upcoming-runs/action.yml

```diff
@@ -0,0 +1,54 @@
+name: Fail if an upcoming run is detected
+description: Exits on error when an upcoming run is detected on the same branch, and rerun it if cancelled
+inputs:
+  github-token:
+    description: GitHub Token
+    required: true
+  workflow:
+    description: The workflow file name eg workflow.yml
+    required: true
+
+outputs:
+  exit:
+    description: True if this check produced an exit
+    value: ${{ steps.check.outputs.exit }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get branch name
+      uses: Alfresco/alfresco-build-tools/.github/actions/get-branch-name@469ed8f7c8198c4f8476392d944b2568e4043c5a # v5.13.1
+
+    - name: Check upcoming runs
+      id: check
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+        WORKFLOW: ${{ inputs.workflow }}
+      shell: bash
+      run: |
+        URL="/repos/$GITHUB_REPOSITORY/actions/workflows/$WORKFLOW/runs"
+        echo "Checking runs at $URL for $BRANCH_NAME"
+        RUNS=$(gh api -X GET $URL -F branch=$BRANCH_NAME -F event=pull_request)
```

Comment:
maybe worth having event as input in case workflow is triggered via push

Reply:
done